ludwig/datasets/kdd_appetency/__init__.py

```diff
@@ -0,0 +1,134 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2021 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+from typing import Tuple
+
+import pandas as pd
+
+from ludwig.datasets.base_dataset import BaseDataset, DEFAULT_CACHE_LOCATION
+from ludwig.datasets.mixins.download import ZipandUncompressedFileDownloadMixin
+from ludwig.datasets.mixins.process import MultifileJoinProcessMixin
+from ludwig.constants import SPLIT
+
+
+def load(cache_dir=DEFAULT_CACHE_LOCATION, split=True):
+    dataset = KDDAppetency(cache_dir=cache_dir)
+    return dataset.load(split=split)
+
+class KDDAppetency(ZipandUncompressedFileDownloadMixin, MultifileJoinProcessMixin,
+                   BaseDataset):
+    """
+    The KDD Cup 2009 Appetency dataset
+
+    Additional Details:
+
+    https://www.kdd.org/kdd-cup/view/kdd-cup-2009/Data
+    """
+    def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION, label=None):
+        super().__init__(dataset_name="kdd_appetency", cache_dir=cache_dir)
+
+    def read_file(self, filetype, filename, header=0):
+        if filetype == 'data':
+            file_df = pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=header, sep='\t')
+        elif filetype == 'labels':
+            file_df = -pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=None)[0]
+        elif filetype == 'txt':
+            file_df = pd.read_csv(
+                os.path.join(self.raw_dataset_path, filename), header=None)
+        return file_df
+
+    def process_downloaded_dataset(self, header=0):
+        super().process_downloaded_dataset()
+        unprocessed_df = pd.read_csv(os.path.join(self.processed_dataset_path,
```

Comment:
this should likely be `raw_dataset_path`

Reply:
I believe `processed_dataset_path` works fine as `process_download_dataset` has been called to concatenate the train and test data into the csv file.